### PR TITLE
feat(lidar_centerpoint): add IoU-based NMS

### PR DESCRIPTION
**Pull request by yukke42**
Signed-off-by: yukke42 <yusuke.muramatsu@tier4.jp>

## Description

In lidar_centerpoint, center-distance-based non-maximum-suppression (NMS) has been used to remove overlapped objects, but it sometimes fails to do it especially for large objects, s.t. trucks and trailers. I add iou-based NMS to remove overlapped objects without any affects to the small objects.
And then, I change the `circle_nms_dist_threshold` param not to remove overlapped pedestrians that are close to each other.

before:
Screenshot from 2022-09-22 16-18-34

Screenshot from 2022-09-22 18-08-45

after:

Screenshot from 2022-09-22 16-21-50

Screenshot from 2022-09-22 18-08-06

## Related links

- [TIERIV INTERNAL LINK](https://tier4.atlassian.net/browse/T4PB-21175)

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [x] I've confirmed the [contribution guidelines].
- [x] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [ ] The PR follows the [pull request guidelines].
- [ ] The PR has been properly tested.
- [ ] The PR has been reviewed by the code owners.

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [ ] There are no open discussions or they are tracked via tickets.
- [ ] The PR is ready for merge.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### CODE REVIEW

**Comment by knzo25** [comment]
Unrelated to the current PR, but maybe `DONWLOAD_BASE_URL` to `DOWNLOAD_BASE_URL` as indicated by the spell checker

File: perception/lidar_centerpoint/include/lidar_centerpoint/postprocess/non_maximum_suppression.hpp [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+  std::vector<std::string> target_class_names_{};
+  double search_distance_2d_{};
+  double iou_threshold_{};
+  // double distance_threshold_{};
+};
+
+std::vector<bool> classNamesToBooleanMask(const std::vector<std::string> & class_names)
+{
+  std::vector<bool> mask;
+  mask.resize(8);
```
Comment (knzo25) [inline perception/lidar_centerpoint/include/lidar_centerpoint/postprocess/non_maximum_suppression.hpp:51]: magic number ?

File: perception/lidar_centerpoint/src/node.cpp [hunk 2]
```diff
@@ -14,6 +14,9 @@
 
 #include "lidar_centerpoint/node.hpp"
 
+#include "perception_utils/perception_utils.hpp"
```
Comment (knzo25) [inline perception/lidar_centerpoint/src/node.cpp:17]:
```
#include "perception_utils/perception_utils.hpp"
#include "tier4_autoware_utils/tier4_autoware_utils.hpp"
```
seem to be unnecessary

**Review by knzo25 — CHANGES_REQUESTED** [review]
Just left some nitpick-like comments.
I also confirmed that the PR works as expected.

File: perception/lidar_centerpoint/include/lidar_centerpoint/postprocess/non_maximum_suppression.hpp (hunk 1 above)
Comment (yukke42) [reply perception/lidar_centerpoint/include/lidar_centerpoint/postprocess/non_maximum_suppression.hpp]: chnged at [6caf06a](https://github.com/autowarefoundation/autoware.universe/pull/1935/commits/6caf06a0c52cd1c654b56a835b97105f0f899de0).

File: perception/lidar_centerpoint/src/node.cpp (hunk 2 above)
Comment (yukke42) [reply perception/lidar_centerpoint/src/node.cpp]: removed at [21cb9f5](https://github.com/autowarefoundation/autoware.universe/pull/1935/commits/21cb9f5be1bcd1db097501d7c9f3adb56abb51c4).

**Review by knzo25 — APPROVED** [review]
Other than the typo, LGTM

`Unknown word (DONWLOAD)`

**Comment by yukke42** [comment]
> Other than the typo, LGTM
> 
> `Unknown word (DONWLOAD)`

Thank you. I fixed at [334a4fd](https://github.com/autowarefoundation/autoware.universe/pull/1935/commits/334a4fd5fa802b15b641bee91fa7ca422a36d8ec).

**Pull request merged**